runtime/common/src/apis.rs

```diff
@@ -475,10 +475,17 @@ macro_rules! impl_runtime_apis_plus_common {
 
 			#[cfg(feature = "try-runtime")]
 			impl frame_try_runtime::TryRuntime<Block> for Runtime {
-				fn on_runtime_upgrade() -> Result<(Weight, Weight), sp_runtime::RuntimeString> {
+				fn on_runtime_upgrade() -> (Weight, Weight) {
 					log::info!("try-runtime::on_runtime_upgrade()");
-					let weight = Executive::try_runtime_upgrade()?;
-					Ok((weight, BlockWeights::get().max_block))
+					// NOTE: intentional unwrap: we don't want to propagate the error backwards, 
+					// and want to have a backtrace here. If any of the pre/post migration checks
+					// fail, we shall stop right here and right now.
+					let weight = Executive::try_runtime_upgrade().unwrap();
```

Comment:
`expect` might make sense here.

```Rust
let weight = Executive::try_runtime_upgrade()
  .expect("If any of the pre/post migration checks fail, we shall stop right here and right now.");
```

Reply:
I agree, please use expect instead of unwrap. And more generally, please never use unwrap.

Reply:
Agreed -- this was copied verbatim from `substrate` but a useful error message wouldn't hurt.

Reply:
addressed in 93b52c6